php/libraries/NDB_BVL_Instrument_JSON.class.inc

```diff
@@ -0,0 +1,289 @@
+<?php
+/**
+ * This file contains the JSON Instrument class
+ *
+ * PHP Version 5
+ *
+ * @category Main
+ * @package  Behavioural
+ * @author   Unknown <example@example.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+namespace Loris\Behavioural;
+use \Exception;
+/**
+ * JSON Instrument class
+ *
+ * @category Main
+ * @package  Behavioural
+ * @author   Unknown <example@example.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+class NDB_BVL_Instrument_JSON extends \NDB_BVL_Instrument
+{
+    var $InstrumentType = 'JSON';
+
+
+    /**
+     * Sets up the variables required for a JSON instrument to load
+     *
+     * @param string $commentID The CommentID being loaded
+     * @param string $page      The page being loaded
+     *
+     * @return none
+     */
+    function setup($commentID = NULL, $page = NULL)
+    {
+        $this->commentID     = $commentID;
+        $this->page          = $page;
+        parent::setup($commentID, $page);
+    }
+
+    /**
+     * Returns candidate-related context variables to be accessed during runtime
+     * Allows access to: date of birth, age in months (calculated)
+     * 
+     * @return array containing context variables
+     */
+    function _getContext() {
+        $timepoint = \TimePoint::singleton($this->getSessionID());
+        $candID = $timepoint->getCandID();
+        $candidate = \Candidate::singleton($candID);
+        $dob = $candidate->getCandidateDoB();
+        $dobDT = new \DateTime($dob);
+        $curDate = new \DateTime("now");
+        $interval = date_diff($curDate, $dobDT, true);
+        $ageM = $this->calculateAgeMonths(array('year' => $interval->format("%y"),
+                                         'mon'  => $interval->format("%m"),
+                                         'day'  => $interval->format("%d")));
+        //$ageTEST = $this->getCandidateAge($curDate); //TODO replace the hacky age function with this
+        $gender = $candidate->getCandidateGender();
+        return array(
+            "age_mths" => $ageM,
+            "dob" => $dob,
+            "gender" => $gender
+        );
+    }
+
+    function _getLang() {
+        //$timepoint = \TimePoint::singleton($this->getSessionID());
+        //$candID = $timepoint->getCandID();
+        //$candidate = \Candidate::singleton($candID);
+        //$lang = $candidate->getCandidateLanguage();
+        $lang = "en-ca";
+        return $lang ? $lang : "en-ca";
+    }
+
+
+    function _getInstrumentData($db) {
+        $result = $db->pselect(
+            "SELECT * FROM $this->table WHERE CommentID=:CID",
+            array('CID' => $this->getCommentID())
+        );
+        return $result ? $result[0] : array();
+    }
```

Comment:
should this be in the base instrument class?

The ternary also looks weird/broken here.. pselect returns an empty array if the result returns nothing anyways, so I don't see why it can't just return $result;